### PR TITLE
chore: Dependency bumps

### PR DESCRIPTION
**Pull request by sbernauer**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -12,15 +12,15 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
-      - uses: dtolnay/rust-toolchain@1.75.0
+      - uses: dtolnay/rust-toolchain@1.79.0
```
Comment (Techassi) [inline .github/workflows/build.yaml:15]: suggestion: This value could be moved into an env variable, so that in the future it only needs to be adjusted in a single place.

File: trino-lb/src/trino_client/cluster_info.rs [hunk 2]
```diff
@@ -34,6 +34,7 @@ pub enum Error {
     },
 }
 
+#[allow(dead_code)]
```
Comment (Techassi) [inline trino-lb/src/trino_client/cluster_info.rs:37]: question: Why is this needed? Shouldn't we remove dead code instead of allowing it here?

**Review by Techassi — COMMENTED** [review]
LGTM, just two small comments.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (sbernauer) [reply .github/workflows/build.yaml]: Please have a look at dfb3affaa9ca681b7e96b67c5b63be471b5b821b

File: trino-lb/src/trino_client/cluster_info.rs (hunk 2 above)
Comment (sbernauer) [reply trino-lb/src/trino_client/cluster_info.rs:37]: This is the HTTP json response we get from the trino cluster when asking for it's state. We current#y don't read all it's fields, but on the other hand I already "reverse engineered" them, so I though there is value in keeping them in case we need them later on. But the compiler is too smart.
No strong opinion on this

Comment (Techassi) [reply trino-lb/src/trino_client/cluster_info.rs:37]: Nope, that sounds fine. I was just curious. We can leave it as is :)

**Review by Techassi — APPROVED** [review]
